### ISSUE TITLE
used-before-assignment and undefined-loop-variable false positives with regard to function annotation

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#1082]
Given a function such as the following, we are getting used-before-assignment and undefined-loop-variable for the annotation, while we should retrieve an undefined-variable message, since the variable is actually not defined there.

```

def test(a:x):
   if x == 2:
      for x in [1, 2]:
          pass
```
